soc/arm/microchip_mec/mec172x/reg/mec172x_ecia.h

```diff
@@ -100,6 +103,18 @@
 #define MCHP_ECIA_BLK_ENCLR_OFS		0x204u
 #define MCHP_ECIA_BLK_ACTIVE_OFS	0x208u
 
+/* 8 <= n <= 26 */
+#define MCHP_GIRQ_SRC_ADDR(n) \
+	((MCHP_ECIA_ADDR + 0x00u) + (((uint32_t)(n) - 8u) * 0x14u))
+
+#define MCHP_GIRQ_ENSET_ADDR(n) \
+	((MCHP_ECIA_ADDR + 0x04u) + (((uint32_t)(n) - 8u) * 0x14u))
+
+/* 8 <= n <= 26 corresponding to GIRQ08, GIRQ09, ..., GIRQ26 */
+#define MCHP_GIRQ_SRC(n) REG32(MCHP_GIRQ_SRC_ADDR(n))
+
```

Comment:
for MEC17xx we are not using macros to access GIRQ registers. We have functions in interrupt driver for this purpose.

Reply:
Do you mean this?
MCHP_GIRQ_SRC ==> mchp_xec_ecia_girq_src_clr
MCHP_GIRQ_ENSET ==>mchp_xec_ecia_girq_src_en

Do you have an official version of counter_mchp_xec.c which can be used for both MEC15 and MEC17?
If so, could you please upload that file?